File: 169. Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> hm = new HashMap<>();
        for(int num : nums) hm.put(num, hm.getOrDefault(num, 0) + 1);

        int maxKey = nums[0];           
        int maxCount = hm.get(maxKey);       
        for (Map.Entry<Integer, Integer> ele : hm.entrySet()) {
            if (ele.getValue() > maxCount) {
                maxKey = ele.getKey();
                maxCount = ele.getValue();
            }
        }
        return maxKey;
    }
}

File: 169. Majority Element.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> hm;
        for (int num : nums) {
            hm[num]++; // count frequency
        }

        int maxKey = nums[0];
        int maxCount = hm[maxKey];

        for (auto &ele : hm) {
            if (ele.second > maxCount) {
                maxKey = ele.first;
                maxCount = ele.second;
            }
        }
        return maxKey;
    }
};
